components/eamxx/src/physics/mam/eamxx_mam_microphysics_process_interface.cpp

```diff
@@ -218,6 +217,29 @@ void MAMMicrophysics::set_grids(
 
     // Diagnostics: H2SO4 in-cloud tendencies[kg/kg/s]
     add_field<Computed>("mam4_microphysics_tendency_aqh2so4", vector3d_mid_nmodes, kg / kg / s, grid_name);
+
+    // get_3d_tensor_layout extents are [ncol, gas_pcnst, nq<...>, nlev]
+    FieldLayout tensor3d_gas_tend =
+        grid_->get_3d_tensor_layout(true, {num_gas_aerosol_constituents_, num_gas_tend_},
+                                    {"num_gas_aerosol_constituents", "num_gas_tend_"});
+    FieldLayout tensor3d_gas_tend_cw =
+        grid_->get_3d_tensor_layout(true, {num_gas_aerosol_constituents_, num_gas_tend_cw_},
+                                    {"num_gas_aerosol_constituents", "num_gas_tend_cw_"});
```

Comment:
These should be dimensioned (icol, gas_pcnst,lev). We should not expose `num_gas_tend_` as it is an internal MAM4 array dimension.